x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -258,6 +273,29 @@ static DeprecationIssue tooManyFieldsCheck(IndexMetaData indexMetaData) {
         return null;
     }
 
+    static DeprecationIssue deprecatedDateTimeFormat(IndexMetaData indexMetaData) {
+        List<String> fields = new ArrayList<>();
+
+        fieldLevelMappingIssue(indexMetaData, ((mappingMetaData, sourceAsMap) -> fields.addAll(
+            findInPropertiesRecursively(mappingMetaData.type(), sourceAsMap,
+                IndexDeprecationChecks::isDateFieldWithDeprecatedPattern
+                , IndexDeprecationChecks::formatDateField))));
+
+
+        if (fields.size() > 0) {
+            return new DeprecationIssue(DeprecationIssue.Level.WARNING,
+                "Date time field format likely contain deprecated pattern",
```

Comment:
I'd be a bit more descriptive here, maybe something like

```suggestion
                "Date field format uses patterns which may change meaning in 7.0",
```

I'm not sure that's quite right either though, because some of them are actually deprecated.

Reply:
I think it should be ok to only imply that pattern may change meaning. For a lot of people who used 'y' in joda 'y' is still ok in java (just the negative year change). So for these people it 'may' change meaning, but I suppose most of the time is ok.

Reply:
It sounds from your comment like you're okay with this change, but it hasn't been made yet - am I misinterpreting? Just making sure this wasn't forgotten.

Reply:
somehow I applied the change from your other comment, but forgot about this one. Applied the change now. Thanks!